### PR TITLE
The ability to delete an ASG via asgard.

### PR DESCRIPTION
**Pull request by macdiesel**

### CODE REVIEW

File: tubular/asgard.py [hunk 1]
```diff
@@ -294,6 +295,33 @@ def disable_asg(asg):
...
+    curl -d "name=helloworld-example-v004" http://asgardprod/us-east-1/cluster/delete
+
+    Arguments:
+        asg(str): The name of the asg to delete.
+
+    Returns:
+        None: When the asg has been deleted.
+
+    Raises:
+        TimeoutException: If the task to enable the ASG fails...
```
Comment (feanil) [inline tubular/asgard.py]: This should be `If the task to delete ASG fails...`

File: tubular/asgard.py [hunk 2]
```diff
@@ -294,6 +295,33 @@ def disable_asg(asg):
...
+        BackendError: If asgard was unable to disable the ASG
```
Comment (feanil) [inline tubular/asgard.py]: `If asgard was unable to delete the ASG`

**Comment by feanil** [comment]
Two documentation issues but otherwise 👍

**Comment by macdiesel** [comment]
I should have tagged this as WIP as I wasn't ready for review just yet but thanks for the catch.

File: tubular/asgard.py (hunk 1 above)
Comment (macdiesel) [inline tubular/asgard.py]: done

File: tubular/asgard.py (hunk 2 above)
Comment (macdiesel) [inline tubular/asgard.py]: done

**Comment by macdiesel** [comment]
@feanil @doctoryes PTAL. Added the ability to delete an ASG. When calling this method by default it will check to see if the ASG is still active. If this is the case an exception will be raised.

**Comment by feanil** [comment]
The other case that we should handle is if they give us the name of an ASG that has already been deleted.

**Comment by feanil** [comment]
👍  once we handle that case.

**Comment by macdiesel** [comment]
@feanil This case is already handled. I have updated the docs to make this more clear.

File: scripts/delete-asg.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+@click.command()
+@click.option('--asg_name', envvar='ASG_NAME', help='the name of the Autoscale Group to delete', required=True)
+def delete_asg(asg_name):
+    try:
+        asgard.delete_asg(asg_name, True)
+    except Exception, e:
+        traceback.print_exc()
+        click.secho("Error Deploying AMI: {0}.\nMessage: {1}".format(asg_name, e.message), fg='red')
```
Comment (doctoryes) [inline scripts/delete-asg.py:23]: s/Deploying AMI/deleting ASG/

Comment (macdiesel) [inline scripts/delete-asg.py:23]: Bad copy/pasta. Will fix in the next PR.

**Pull request merged**